src/test/scala/chisel3/internal/NamespaceSpec.scala

```diff
@@ -78,4 +78,10 @@ class NamespaceSpec extends AnyFlatSpec {
     name("x") should be("x_2")
     name("x_0") should be("x_0_1")
   }
+
+  they should "support resolving collisions between <name> and <name>_" in {
+    val namespace = Namespace.empty
+    namespace.name("mouth") should be("mouth")
+    namespace.name("mouth_") should be("mouth_")
```

Comment:
What happens if we ask for `mouth` and `mouth_` again? Might be good to include in the test.

Reply:
Added it, creates `mouth_1` and `mouth__1`